src/compiler/checker.ts

```diff
@@ -25684,7 +25939,7 @@ namespace ts {
                 }
             }
             if (restType) {
-                const spreadType = getSpreadArgumentType(args, argCount, args.length, restType, /*context*/ undefined);
+                const spreadType = getSpreadArgumentType(args, argCount, args.length, restType, /*context*/ undefined, checkMode);
                 const errorNode = reportErrors ? argCount < args.length ? args[argCount] : node : undefined;
                 if (!checkTypeRelatedTo(spreadType, restType, relation, errorNode, headMessage, /*containingMessageChain*/ undefined, errorOutputContainer)) {

```

Comment:
a bit more detail from debugging the VS Code crash again: 

```ts
declare function crash(...args: unknown[]): any;
export function gift(us: unknown[]) {
  crash(...us);
}
```

at this point, spreadType is the reference object type for `us`, whose target is an anonymous object type. Neither type has a symbol. When resolveStructuredTypeMembers is called on spreadType inside of checkTypeRelatedTo (relating `us` to `args`), resolveTypeReferenceMembers calls resolveDeclaredMembers spreadType.target. resolveDeclaredMembers assumes that spreadType.target has a symbol, which it does not.

Reply:
The only reason we're in this block is that `restType` is defined. A standalone run has `restType` undefined because it's not an array type. That must mean that VS Code's compile messes up `globalArrayType` in some way -- changing the program to use `Array<unknown>` avoids the crash:

```ts
declare function crash(...args: Array<unknown>): any;
export function gift(us: unknown[]) {
  crash(...us);
}
```

Reply:
Ah, maybe it only happens when the global array type couldn't be located. We then default it to an empty object type--which has no symbol.

Reply:
Yeah, that appears to be the case. When I compile the repro with `tsc -nolib` it does indeed crash.

Reply:
That's what I confirmed in the debugger. Is the existing code in the checker resilient to missing the Array type?

Reply:
Now fixed.